app/routes/goal_routes.py

```diff
@@ -0,0 +1,82 @@
+from flask import Blueprint, jsonify, abort, make_response, request
+from app.models.goal import Goal
+from app.models.task import Task
+from app.routes.helper_routes import validate_object
+from app import db
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals():
+    
+    sort_query = request.args.get("sort")
+
+    if sort_query == "asc":
+        goals = Goal.query.order_by(Goal.title.asc())
+    
+    elif sort_query == "desc":
+        goals = Goal.query.order_by(Goal.title.desc())
+    else:
+        goals = Goal.query.all()
+        
+    goal_dict = [goal.to_dict() for goal in goals]
+    return jsonify(goal_dict), 200
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def single_goal(goal_id):
+    goal = validate_object(Goal,goal_id)
+    return jsonify({"goal": goal.to_dict()}), 200
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_object(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
+
+    db.session.commit()
+    
+    return jsonify({"goal":goal.to_dict()}), 200
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_single_goal(goal_id):
+    goal = validate_object(Goal, goal_id)
+
+    db.session.delete(goal)    
+    db.session.commit()
```

Comment:
Great job using `db.session.<method-name>.` Session gives us access to the following:

* `db` is our app’s instance of SQLAlchemy.
* `session` represents our active database connection.
* By referencing `db.session` we can use SQLAlchemy’s methods to perform tasks like committing a change to a model, storing a new record of a model, and deleting a record.

By referencing `db.session.add()` you are able to use the SQLAlchemy method to store a new record of the Goal model. ⭐